src/BlockTypes/ProductQuery.php

```diff
@@ -474,15 +474,29 @@ function( $acc, $attribute ) {
 	/**
 	 * Return queries that are generated by query args.
 	 *
+	 * @param array $query Query args.
+	 *
 	 * @return array
 	 */
-	private function get_queries_by_applied_filters() {
-		return array(
+	private function get_queries_by_applied_filters( $query ) {
+		$result = array(
 			'price_filter'        => $this->get_filter_by_price_query(),
 			'attributes_filter'   => $this->get_filter_by_attributes_query(),
 			'stock_status_filter' => $this->get_filter_by_stock_status_query(),
 			'rating_filter'       => $this->get_filter_by_rating_query(),
 		);
+
+		// "Product Categories" & "Product Tags" could be provided using "Filters" ToolsPanel available in Inspector Controls.
+		if ( isset( $query['tax_query'] ) ) {
+			$result['tax_query'] = $this->get_filter_by_product_categories_or_tags_query( $query['tax_query'] );
+		}
```

Comment:
@sunyatasattva @gigitux Can you please take a look at this PR? I have a query regarding these lines of code.

As I am not sure what else could be there in `$query['tax_query']`, therefore I have created `get_filter_by_product_categories_or_tags_query` function to extract queries related to only Product categories & Product tags.

I am wondering if we could use `$query['tax_query']` as it is? Is there any possibility of extra items in `$query['tax_query']` which we won't need & therefore we need to extract only what we need?

Reply:
Also, I can see that there is [get_global_query](https://github.com/woocommerce/woocommerce-blocks/blob/1954437c4616fedd27a2c18b41134b5ef1cfd0b4/src/BlockTypes/ProductQuery.php#L761) function. But code of this function only runs if `is_custom_inherit_global_query_implementation_enabled` is true. 

Only this function extracts `Keyword` which could be provided using "Filters" ToolsPanel available in Inspector Controls. Don't we need `Keyword` when is_custom_inherit_global_query_implementation_enabled is false?

Reply:
>I am wondering if we could use $query['tax_query'] as it is 🤔 Is there any possibility of extra items in $query['tax_query'] which we won't need & therefore we need to extract only what we need((For example, only Product categories & tags))?

I don't recall exactly the reasoning behind it, as @gigitux has worked on that part of the code, but I believe it is intentional. Perhaps something to do with the interaction with filters blocks if we take `tax_query` as it is?

Or maybe it was a prophylactic measure to enable the `custom_inherit_global_query`, which we eventually want to switch to.

> Don't we need Keyword when is_custom_inherit_global_query_implementation_enabled is false?

Yes, we definitely do. It must have been an oversight we didn't catch, as we were focused on implementing the custom inherit, before the scope was changed.

Reply:
I don't remember very well, tbh 😭. 

In any case, I think that we should not use the `$query['tax_query'] as it is` to be ready when we will switch to the custom inherit query logic. 

We already have the logic[ to merge the queries](https://github.com/woocommerce/woocommerce-blocks/blob/4eda72955108774835946da7974b92aebf77c59b/src/BlockTypes/ProductQuery.php/#L196-L197).

Reply:
Hi @gigitux 👋 

> In any case, I think that we should not use the $query['tax_query'] as it is to be ready when we will switch to the custom inherit query logic.

Got it, then the changes I made in this PR looks good to me. In short, I created `get_filter_by_product_categories_or_tags_query` function to extract `product_tag` & `product_cat` from the `tax_query` as you can see [here](https://github.com/woocommerce/woocommerce-blocks/blob/1954437c4616fedd27a2c18b41134b5ef1cfd0b4/src/BlockTypes/ProductQuery.php#L832-L869). Also, I extracted the keyword [here](https://github.com/woocommerce/woocommerce-blocks/blob/1954437c4616fedd27a2c18b41134b5ef1cfd0b4/src/BlockTypes/ProductQuery.php#L786-L788). Please let me know if you notice any problem with this implementation. 

> We already have the logic[ to merge the queries](https://github.com/woocommerce/woocommerce-blocks/blob/4eda72955108774835946da7974b92aebf77c59b/src/BlockTypes/ProductQuery.php/#L196-L197).

Right, but while calling `merge_queries` function, we were not passing `tax_query` for product categories, product tags & keyword. I made the changes in this PR to pass these to `merge_queries` function. 

Please let me know if I misunderstood something & if you see any problem with changes I made in this PR.

Reply:
I would create a dedicated function for the filters applied on the editor side. The documentation about `get_queries_by_applied_filters` says:

```
Return queries that are generated by query args.
```

Reply:
@gigitux That makes sense. There are two sections of filters in Editor i.e. `Advance Filters` & `Filters`  (see screenshot)
![image](https://user-images.githubusercontent.com/16707866/218707003-00a2a6fe-aa16-4718-80fe-8f33f0b0372e.png)

As per my understanding, we are creating queries for Advance filters in [get_queries_by_attributes](https://github.com/woocommerce/woocommerce-blocks/blob/1954437c4616fedd27a2c18b41134b5ef1cfd0b4/src/BlockTypes/ProductQuery.php#L508) (which isn't obvious from the name of function). 

I am thinking to do one of the following:
1. Maybe we can rename this function to `get_filter_queries` & return queries for `Advance Filters` & `Filters` from this function.
2. Rename this function to `get_advance_filter_queries` to make it more readable and clear. And, create new function called `get_filter_queries` which will return queries for product categories, product tags & keyword i.e. all the filters. Then pass `get_filter_queries()` as argument [here](https://github.com/woocommerce/woocommerce-blocks/blob/1954437c4616fedd27a2c18b41134b5ef1cfd0b4/src/BlockTypes/ProductQuery.php#L159-L165).

Which of these you think would be better approach?

Reply:
> That function generates queries by the custom attributes that the merchant sets via the editor.

Got it, maybe we can rename the function to `get_queries_by_custom_attributes` just to make it more readable & clear. What do you think? 🤔

> That function generates queries by the custom attributes that the merchant sets via the editor. Furthermore, I think that there should not be a relationship between the UI and the name of the function of the backend.
So, I would create another function to create a valid query object starting from the $tax_query object.

That makes sense. I will create a new function. Thanks for the feedback 🙌🏻

Reply:
> What do you think?

It makes sense! 👍 

> That makes sense. I will create a new function. Thanks for the feedback 🙌🏻

💪

Reply:
@gigitux I made the changes in [this commit](https://github.com/woocommerce/woocommerce-blocks/pull/8377/commits/db26bdb7d86434f367bf854e2d9b56141154a17b). Would you like to have a quick look?